File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06IO1/BridgePattern/BridgePattern/ShapeFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    public abstract class ShapeFactory
    {
        private delegate Shape ShapeMaker(string parameters);
        private static Dictionary<string, ShapeData> map = new Dictionary<string, ShapeData>()
        {
            {"Circle",    new ShapeData((parameters) => new ShapeCircle(parameters),"100 100 90")    },
            {"Triangle",  new ShapeData((parameters) => new ShapeTriangle(parameters),"10 10 190 95 95 190")  },
            {"Rectangle", new ShapeData((parameters) => new ShapeRectangle(parameters),"10 10 190 190") },
        };
        public static string[] Kinds = map.Keys.OrderBy(kind => kind).ToArray();
        public static Shape make(string kind,string parameters) { return map[kind].maker(parameters); }
        public static string paremeters(string kind) { return map[kind].parameters; }
        private class ShapeData
        {
            public ShapeMaker maker { get; private set; }
            public string parameters { get; private set; }
            public ShapeData(ShapeMaker maker, string parameters)
            {
                this.maker = maker;
                this.parameters = parameters;
            }
        }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO3/BridgePattern/src/ShapeFactory.java

import java.util.HashMap;
import java.util.List;
import java.util.stream.Collectors;

public abstract class ShapeFactory
{
    private static interface ShapeMaker { public ShapeBase make(String text); }
    private static final HashMap<String,ShapeMaker> map=new HashMap<String,ShapeMaker>()
    {
        {
            put("Circle",(text)->new ShapeCircle(text));
            put("Rectangle",(text)->new ShapeRectangle(text));
            put("Triangle",(text)->new ShapeTriangle(text));
        }
    };
    public static final List<String> keys=map.keySet().stream().sorted().collect(Collectors.toList());
    public static ShapeBase make(String key,String text) { return map.get(key).make(text); }
}
